src/Ring.cc

```diff
@@ -79,13 +79,45 @@ std::pair<double, int> Ring::computeOptimalRingParametersRectangle(double module
 }
 
 
-void Ring::buildModules(EndcapModule* templ, int numMods, double smallDelta) {
+void Ring::buildModules(EndcapModule* templ, int numMods, double smallDelta, double phiShift) {
   double alignmentRotation = alignEdges() ? 0.5 : 0.;
+  double deltaPhiNom = 2.*M_PI/numMods;
+  double deltaPhiLarge = deltaPhiNom+2*phiShift;
+  double nominalZRot = 0.;
+  if (deltaPhiLarge!=deltaPhiNom){ deltaPhiNom = deltaPhiNom-(4*phiShift/(numMods-2));}
   for (int i = 0, parity = smallParity(); i < numMods; i++, parity *= -1) {
     EndcapModule* mod = GeometryFactory::clone(*templ);
+    if (moduleNode.count(i) > 0){ mod->store(moduleNode.at(i));}
     mod->myid(i+1);
-    mod->rotateZ(2.*M_PI*(i+alignmentRotation)/numMods); // CUIDADO had a rotation offset of PI/2
+    std::cout<<"------begin module printout--------"<<std::endl;
+    if(mod->propertyTree().get<double>("twistAng",-99) > -99 ){//Need to do this before further determining phi/z of the module
```

Comment:
I would rather choose -999which is not a valid angle

Actually at a point we should consider switching these return values to using std::optional<>, but that would require a lot of work...